Worker/src/Common/Adapter/S3Adapter.cs

```diff
@@ -116,9 +116,15 @@ await r.WriteResponseStreamToFileAsync(materializedFileName,
                                              false,
                                              cancellationToken)
              .ConfigureAwait(false);
-
-      File.Move(materializedFileName,
-                targetFileName);
+      try
+      {
+        File.Move(materializedFileName,
+                  targetFileName);
+      }
+      catch (IOException ioException)
+      {
+        // File already exists
+      }
```

Comment:
Is there something finer than that ? It looks a bit wide to me

Reply:
https://learn.microsoft.com/fr-fr/dotnet/api/system.io.file.move?view=net-8.0#system-io-file-move(system-string-system-string) It's the raised exception

Reply:
yes but this can be raised in other cases too. https://learn.microsoft.com/en-us/dotnet/api/system.io.ioexception?view=net-9.0#remarks